src/page-modules/departures/details/estimated-call-rows.tsx

```diff
@@ -0,0 +1,199 @@
+              exit={{ opacity: 0 }}
+              transition={{
+                delay: i * 0.05,
+              }}
+            >
+              <EstimatedCallRow
+                call={call}
+                mode={mode}
+                collapseButton={
+                  call.metadata.isStartOfServiceJourney ? collapseButton : null
+                }
+              />
+            </motion.div>
+          ))}
+        </div>
+      )}
+
+      <div className={style.callRows__container}>
+        {estimatedCallsToShow.map((call, i) => (
+          <motion.div
+            key={`${call.quay.id}-${call.aimedDepartureTime}`}
+            initial={{ opacity: 0 }}
+            animate={{ opacity: 1 }}
+            exit={{ opacity: 0 }}
+            transition={{
+              delay: i * 0.015,
+            }}
+          >
+            <EstimatedCallRow call={call} mode={mode} collapseButton={null} />
+          </motion.div>
+        ))}
+      </div>
+    </div>
+  );
+}
+
+type EstimatedCallRowProps = {
+  call: EstimatedCallWithMetadata;
+  mode: TransportModeType;
+  collapseButton: JSX.Element | null;
+};
+function EstimatedCallRow({
+  call,
+  mode,
+  collapseButton,
+}: EstimatedCallRowProps) {
+  const { t } = useTranslation();
+  const { group, isStartOfGroup, isEndOfGroup } = call.metadata;
+  const isBetween = !isStartOfGroup && !isEndOfGroup;
+  const iconColor = useTransportationThemeColor({
+    mode: group === 'trip' ? mode : 'unknown',
+  });
+
+  return (
+    <div className={style.rowContainer}>
+      <DecorationLine
+        hasStart={isStartOfGroup}
+        hasCenter={isBetween}
+        hasEnd={isEndOfGroup}
+        color={iconColor.backgroundColor}
+      />
+      <Row
+        rowLabel={
+          <DepartureTime
+            aimedDepartureTime={call.aimedDepartureTime}
+            expectedDepartureTime={call.expectedDepartureTime}
+            realtime={call.realtime}
+            isStartOfServiceJourney
+          />
+        }
+        alignChildren={
+          isStartOfGroup ? 'flex-start' : isEndOfGroup ? 'flex-end' : 'center'
+        }
+        isBetween={isBetween}
+        href={`/departures/${call.quay.stopPlace.id}`}
+      >
+        <Typo.p textType="body__primary">{call.quay.name}</Typo.p>
+        {!call.forAlighting && !call.metadata.isStartOfServiceJourney && (
+          <Typo.p textType="body__secondary" className={style.boardingInfo}>
+            {t(PageText.Departures.details.messages.noAlighting)}
+          </Typo.p>
+        )}
+        {!call.forBoarding && !call.metadata.isEndOfServiceJourney && (
+          <Typo.p textType="body__secondary" className={style.boardingInfo}>
+            {t(PageText.Departures.details.messages.noBoarding)}
+          </Typo.p>
+        )}
+      </Row>
+      {collapseButton}
+    </div>
+  );
+}
+
+type RowProps = PropsWithChildren<{
+  rowLabel?: React.ReactNode;
+  alignChildren?: 'flex-start' | 'flex-end' | 'center';
```

Comment:
We need this configurable as a prop?

Reply:
We need to know if the items should be aligned at the start (first quay in a group) or at the end (last quay in a group), or be centered. Alternatively, we could change it and add the props `isStartOfGroup` and `isEndOfGroup` if you think that is better

Reply:
I see now, that's fine. :)